server/server.js

```diff
@@ -1,15 +1,83 @@
+require("dotenv").config();
 const express = require("express");
 const app = express();
 const port = process.env.PORT || 5000;
+const cors = require("cors");
+app.use(cors());
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+const videos = require("./exampleresponse.json");
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+app.use(express.json());
 
 // GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+app.get("/videos", (req, res) => {
+  res.status(200).json(videos);
+});
+
+app.post("/videos", (req, res) => {
+  const { title, url } = req.body;
+  if (!title || !url || !url.startsWith("https://www.youtube.com")) {
```

Comment:
Your client had different validation than your server, for what URLs are allowed - this may be a problem!

Reply:
I have added  a URL validation in my level 300 server.js today.